iree/compiler/InputConversion/MHLO/test/flatten_tuples_in_cfg.mlir

```diff
@@ -0,0 +1,32 @@
+// RUN: iree-opt -split-input-file --iree-mhlo-flatten-tuples-in-cfg -canonicalize %s | IreeFileCheck %s
+// We rely on canonicalization to cancel out tuple/get_element operations, so
+// we do this as slightly more than a unit test of the pass only.
```

Comment:
I don't understand what this means. Ah after reading again I see you mean that's why we canonicalize. How about this:

```suggestion
// we test this pass followed by canonicalization rather than the pass in isolation.
```

But also (not this PR), why don't we just declare those ops illegal/canonicalize within the pass?

Reply:
Yeah, I'm just driving the forklift on this one. I'm secretly hopeful that a couple of upstream passes subsume all of this.

Reply:
Left a TODO.